ethsigner/core/src/main/java/tech/pegasys/ethsigner/core/requesthandler/sendtransaction/DownstreamPathCalculator.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.ethsigner.core.requesthandler.sendtransaction;
+
+import java.util.StringJoiner;
+
+public class DownstreamPathCalculator {
+
+  private final String httpDownstreamPathPrefix;
+
+  public DownstreamPathCalculator(final String httpDownstreamPathPrefix) {
+    this.httpDownstreamPathPrefix = httpDownstreamPathPrefix;
+  }
+
+  public String calculateDownstreamPath(final String receivedRequestUri) {
+    // This function is required to join httpDownStreamPathPrefix with receivedRequestURI
+    // remove any duplicate "/"
+    // Not have an ending "/" if the path
+    // assumes httpdownstreamPathPrefix does not include =,?,&
+    // has a leading "/" as required by vertx
+
+    final StringJoiner joiner = new StringJoiner("/", "/", "");
+    joiner.add(httpDownstreamPathPrefix);
+    joiner.add(receivedRequestUri);
+    String result = joiner.toString();
```

Comment:
Don't think there is much benefit to using StringJoiner in this case. String concatenation would be simpler i.e. `String result = "/" + httpDownstreamPathPrefix + "/" + receivedRequestUri;`

Reply:
done